core/src/io/anuke/mindustry/input/InputHandler.java

```diff
@@ -792,13 +792,21 @@ public void drawArrow(Block block, int x, int y, int rotation){
 
     public void drawArrow(Block block, int x, int y, int rotation, boolean valid){
         Draw.color(!valid ? Pal.removeBack : Pal.accentBack);
+        if(Core.input.keyDown(Binding.dash) && valid){
```

Comment:
I would put this into the ternary statement, since it's part of the color assignment: `Core.input.keyDown(Binding.dash) && valid ? Pal.upgradeBack : !valid ? Pal.removeBack : Pal.accentBack`. Similarly for the back color.